src/core/ext/xds/xds_transport_grpc.cc

```diff
@@ -205,15 +205,18 @@ void GrpcXdsTransportFactory::GrpcXdsTransport::GrpcStreamingCall::
   self->recv_message_payload_ = nullptr;
   self->event_handler_->OnRecvMessage(StringViewFromSlice(response_slice));
   CSliceUnref(response_slice);
-  // Keep reading.
+}
+
+void GrpcXdsTransportFactory::GrpcXdsTransport::GrpcStreamingCall::
+    StartRecvMessage() {
   grpc_op op;
   memset(&op, 0, sizeof(op));
   op.op = GRPC_OP_RECV_MESSAGE;
-  op.data.recv_message.recv_message = &self->recv_message_payload_;
-  GPR_ASSERT(self->call_ != nullptr);
+  op.data.recv_message.recv_message = &recv_message_payload_;
+  GPR_ASSERT(call_ != nullptr);
   // Reuses the "OnResponseReceived" ref taken in ctor.
```

Comment:
Once you make the change that I suggested elsewhere to require the `XdsClient` to call `StartRecvMessage()` even for the initial read, I think this will need to take a new ref:

```
Ref(DEBUG_LOCATION, "OnResponseReceived").release();
```

And we'll need `OnResponseReceived()` to unconditionally unref.